File: leetcode/Python/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        sol, sz, cur, dq= [], 0, None, collections.deque()
        if root is None:
            return 
        dq.append(root)
        while dq:
            t, sz= [], len(dq)
            for _ in range(sz):
                cur=dq.popleft()
                t.append(cur.val)
                if cur.left:
                    dq.append(cur.left)
                if cur.right:
                    dq.append(cur.right)
            sol.append(t)
        return sol

File: leetcode/C++/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        int sz=0;
        vector<vector<int>> sol;
        queue<TreeNode*> q;
        TreeNode* cur;
        if(!root) return sol;
        q.push(root);
        while(!q.empty()){
            sz=q.size();
            vector<int> v;
            for(int i=0; i<sz; ++i){
                cur=q.front(); q.pop();
                v.push_back(cur->val);
                if(cur->left) q.push(cur->left);
                if(cur->right) q.push(cur->right);
            }
            sol.push_back(v);
        }
        return sol;
    }
};